stl/inc/algorithm

```diff
@@ -989,6 +987,34 @@ namespace ranges {
                 }
                 // If we get here, _Count > 1, initial elements do not match, and final elements do not match.
 
+                const auto _ProjectedPred = [&]<class _Ty1, class _Ty2>(_Ty1&& _Left, _Ty2&& _Right) {
+                    return _STD invoke(_Pred, _STD invoke(_Proj1, _STD forward<_Ty1>(_Left)),
+                        _STD invoke(_Proj2, _STD forward<_Ty2>(_Right)));
+                };
+
+                for (;;) {
+                    _TrimResult _Res = _Trim_reversed(_First1, _Final1, _First2, _Final2, _ProjectedPred);
+                    if (_Res != _TrimResult::_KeepTrimming) {
+                        if (_Res == _TrimResult::_Break) {
+                            break;
+                        }
+                        return _Res == _TrimResult::_ReturnTrue;
+                    }
+
+                    _Res = _Trim_equal(_First1, _Final1, _First2, _Final2, _ProjectedPred);
+                    if (_Res != _TrimResult::_KeepTrimming) {
+                        if (_Res == _TrimResult::_Break) {
+                            break;
+                        }
+                        return _Res == _TrimResult::_ReturnTrue;
+                    }
+                }
+
+                ++_Final1;
+                ++_Final2;
+                // If we get here, initial elements do not match, final elements do not match, and ranges have length
+                // at least 2 and at most _Count

```

Comment:
Nitpick: Since we're going to the trouble to add this comment, should we also mention explicitly that we know that we've also checked the prefix of one range against the suffix of the other?  The way the comment is phrased now, it sounds like the matching prefixes and matching suffixes have been removed.

Reply:
I've added "before and after" comments to clarify this, including a diagram to illustrate all of the elements that we've compared.